### ISSUE TITLE
New Tab on Open + All Tabs missing on re-open during same session.

### ISSUE THREAD

**Issue by Tyrewt** [Drive4ik/simple-tab-groups#451]
Since the last update, opening the browser will start on a New Tab which isn't associated with any Group. Closing the Tab will then open an existing Tab currently saved within a Group and everything functions as normal until next session. On next session, this behavior will repeat itself.

However, if you close and re-open the browser within the same session, after a brief pause the saved Tab will load where you left off, but then all saved Tabs are lost from the STG Groups Menu except from the current group which incorrectly reports 15 Tabs.

[Screenshots showing STG Groups Menu once the browser is first opened vs re-opened during the same session](https://imgur.com/a/aFsLorS).

STG 4.0.1.2
Firefox 68.0.2 (Linux)

**Comment by TillDeathCoder** [comment]
I have an alike problem. After closing my browser all my tabs are missing and I need to restore them manually from bookmarks.

**Comment by Tyrewt** [comment]
> I have an alike problem. After closing my browser all my tabs are missing and I need to restore them manually from bookmarks.

Not the same issue. Tabs are present after ending session, only after re-opening of the browser within the same session Tabs are lost.

**Comment by eetievent** [comment]
Similar problem for me : all groups except one get 0 tabs when lauching Firefox. Don't know exacly when it happens (mainly when booting in the morning?)

Environment:
The problem appeared when upgrading to Firefox 68.0.2 and STG 4.0.1.2
Ubuntu bionic 18.04


I noticed also the backup size has changed: 37M before upgrading STG, then 22Mo or 4Mo. Restoring tab groups works with the 22Mo backup, but has the same void tab group problem when restoring small backups.
![simpleTabGroupBugGroups0Tabs](https://user-images.githubusercontent.com/19289630/63683857-cb1e7380-c7fb-11e9-86a9-e3961fcdef22.png)

**Comment by eetievent** [comment]
Related to #448 and #442 .

**Comment by TillDeathCoder** [comment]
> 
> 
> > I have an alike problem. After closing my browser all my tabs are missing and I need to restore them manually from bookmarks.
> 
> Not the same issue. Tabs are present after ending session, only after re-opening of the browser within the same session Tabs are lost.

I wrote not the same but alike.

**Comment by eetievent** [comment]
Reinstalled previous version STG 4.0.1.1 (still Firefox 68.0.2 Linux): similar problem, now all groups get no tabs.
To try: last STG with previous Firefox...

**Comment by eetievent** [comment]
Can I send you backup files for testing?

**Comment by Naarty** [comment]
I managed to find a possible cause for this, loading many tabs then exiting firefox seems to trigger it for me, after this trigger it also continues to be broken on this specific firefox profile, you can reimport a backup etc but upon restarting it will display all tab groups as 0.

**Comment by eetievent** [comment]
The reasons seems to be that "Backup files are broken and unusable" #456

**Comment by TillDeathCoder** [comment]
> 
> 
> The reasons seems to be that "Backup files are broken and unusable" #456

But I can restore my tabs from backup files manually. They don't restore automatically after reopening browser.

**Comment by eetievent** [comment]
the restore works also for me but only for old backups (before Firefox 68.0.2). The  newer backups are much smaller (4M instead of 22M) and get restored with all tabs in 1 group only.

**Comment by Naarty** [comment]
> 
> 
> the restore works also for me but only for old backups (before Firefox 68.0.2). The newer backups are much smaller (4M instead of 22M) and get restored with all tabs in 1 group only.

hmm for me it's different everything works and saves fine if i start a new firefox profile, but if i load too many tabs in the new firefox profile, and then exit firefox, it's gets deleted but if i don't load many tabs it works fine, by many tabs i mean 50-150 tabs.

**Comment by Tyrewt** [comment]
Now using Firefox 69.0 issue remains as initially reported.

**Comment by eetievent** [comment]
With Firefox 69.0 (3 september), it works again for me, the tab groups are correctly populated when starting
(note that backups are still smaller, however they seem to grow everyday)

**Comment by Drive4ik** [comment]
Hello. Guys, keep in mind that version 4 no longer saves tabs in its settings. They remain in the browser session. And the addon does not close the tabs except you yourself do not click to close the tab or delete the group.
Perhaps this is a problem in restoring the browser session. 
For example, every 3-4 years I transfer all my bookmarks and add-ons to a new Firefox profile. Because there are errors during the migration of versions and there may be problems such that there will be no more lack of knowledge. Try to repeat this problem in a new Firefox profile with just this addon.

**Comment by BenjamenMeyer** [comment]
For a while, I would close/open Firefox and find all my tabs in the "hidden group" unassociated with any tab group. Now, I'm losing them entirely. STG isn't even remembering which tabs belong to the tab groups.

I'm filing this here because it is similar across the history of the issue. It wasn't a huge deal until it lost all my tabs entirely. This is now almost making tab groups unusable.

@drive4ik not sure what the underlying cause is, but tab groups are critical to my FF workflow and tab management.

FWIW - I'm on Mac OS 10.14 Mojave; not sure when I'll get the 10.15 Catalina upgrade. FF 69.01. Upgraded now to FF 69.03 so will see what happens on next restart.

**Comment by Drive4ik** [comment]
@BenjamenMeyer Do you enable `restore session` in browser options?

**Comment by BenjamenMeyer** [comment]
@Drive4ik for clarity - here's the settings:
![Screen Shot 2019-10-15 at 12 19 03 PM](https://user-images.githubusercontent.com/1074110/66850082-158fb700-ef46-11e9-9ddf-5e8cba93e7a7.png)
![Screen Shot 2019-10-15 at 12 18 29 PM](https://user-images.githubusercontent.com/1074110/66850083-158fb700-ef46-11e9-8bfa-3238169cadd3.png)

I had to double check b/c I usually have that stuff enabled but wanted to verify.
Do note I am also using some pinned tabs in case that matters.

**Comment by Drive4ik** [comment]
That is, it turns out that whenever you start Firefox, all your groups are reset and all tabs are transferred to "unsynchronized"? Do I understand everything correctly?

Try creating a new Firefox profile and testing there. The connection of the tab with the group is saved in the Firefox session, maybe you turned on/off some setting that is responsible for this, in addition to the "restore session".
There shouldn’t be such a problem on the new profile.

**Comment by BenjamenMeyer** [comment]
Okay - so I restarted today (applied a Mac OS update) and Firefox lost basically the whole thing. I'm beginning to think it may be a FF bug more than STG. I had to go into history and restore a window, which then didn't restore correctly either (missed all my pinned tabs) and was a PITA to complete the restore.... for the moment I'll just see how things keep going, and will try to organize into the tab groups again.

FF use to do really really good at restoring sessions even when multiple tabs were still open. Now it seems to want to close everything out and forget them. 😞 

BTW - I didn't even have the `restore session` on the menu, and `about:sessionrestore` didn't detect any session needing to be restored either.

**Comment by Drive4ik** [comment]
@BenjamenMeyer 
Have you tried restoring the latest STG backup?
You can find it in Firefox download folder or here
![изображение](https://user-images.githubusercontent.com/7843031/66926361-93cb8680-f036-11e9-9504-d6cd773cad28.png)

**Comment by TillDeathCoder** [comment]
> 
> 
> @BenjamenMeyer Do you enable `restore session` in browser options?
It helped me. Thx. It was hell to restore tabs manually ;)

**Comment by BenjamenMeyer** [comment]
I have not... but I will have to remember that the next time. Didn't realize there was an auto-backup happening there - cool beans. Thanks for pointing that out!

**Comment by BenjamenMeyer** [comment]
@Drive4ik  okay - so it did it again - I closed the browser and got the warning about having X number of tabs still open so closed them all - I needed to restart the system. When I restarted Firefox it came up to a different page. Everything was in the history, so I did go there first and restored the window which got back my window state; however, STG still doesn't remember any of the tabs I had assigned to any groups, which for me is important as I use tab groups to organize things and often leave groups hidden for long durations of time until I come back to that subject.
